include/msgpack/v1/detail/cpp11_zone.hpp

```diff
@@ -25,153 +25,108 @@ namespace msgpack {
 /// @cond
 MSGPACK_API_VERSION_NAMESPACE(v1) {
 /// @endcond
-
+    
 class zone {
 private:
     struct finalizer {
-        finalizer(void (*func)(void*), void* data):m_func(func), m_data(data) {}
+        finalizer(void (*func)(void*), void* data, finalizer* next): m_func(func), m_data(data), m_next(next) {}
         void operator()() { m_func(m_data); }
         void (*m_func)(void*);
         void* m_data;
+        finalizer* m_next{};
```

Comment:
```suggestion
        finalizer* m_next;
```

m_next is always initialized by the constructor.

Reply:
Originally I tried to make as few changes to the original as possible, in terms of formatting. But maybe I should also bring the other parts of this file to a common style, like moving '{' to another line and empty lines between methods? And now it's all kind of chaotic.